service/mapper/transitions_test.go

```diff
@@ -0,0 +1,1008 @@
+		{TransactionID: flow.Identifier{0x1, 0x2}},
+		{TransactionID: flow.Identifier{0x3, 0x4}},
+	}
+	testHeader := &flow.Header{
+		ChainID: dps.FlowTestnet,
+		Height:  testHeight,
+	}
+	testTransaction := &flow.TransactionBody{
+		ReferenceBlockID:   flow.Identifier{},
+		Script:             nil,
+		Arguments:          nil,
+		GasLimit:           0,
+		ProposalKey:        flow.ProposalKey{},
+		Payer:              flow.Address{},
+		Authorizers:        nil,
+		PayloadSignatures:  nil,
+		EnvelopeSignatures: nil,
+	}
+	testTransactions := []*flow.TransactionBody{testTransaction}
+
+	testTrie, err := trie.NewMTrie(testRoot)
+	require.NoError(t, err)
+
+	load := &mocks.Loader{
+		CheckpointFunc: func() (*trie.MTrie, error) {
+			return testTrie, nil
+		},
+	}
+
+	chain := &mocks.Chain{
+		RootFunc: func() (uint64, error) {
+			return testHeight, nil
+		},
+		HeaderFunc: func(height uint64) (*flow.Header, error) {
+			return testHeader, nil
+		},
+		CommitFunc: func(height uint64) (flow.StateCommitment, error) {
+			return testNextCommit, nil
+		},
+		EventsFunc: func(height uint64) ([]flow.Event, error) {
+			return testEvents, nil
+		},
+		TransactionsFunc: func(height uint64) ([]*flow.TransactionBody, error) {
+			return testTransactions, nil
+		},
+	}
+
+	index := &mocks.Writer{
+		FirstFunc: func(height uint64) error {
+			return nil
+		},
+		LastFunc: func(height uint64) error {
+			return nil
+		},
+		HeaderFunc: func(height uint64, header *flow.Header) error {
+			return nil
+		},
+		CommitFunc: func(height uint64, commit flow.StateCommitment) error {
+			return nil
+		},
+		EventsFunc: func(height uint64, events []flow.Event) error {
+			return nil
+		},
+		PayloadsFunc: func(height uint64, paths []ledger.Path, value []*ledger.Payload) error {
+			return nil
+		},
+		HeightFunc: func(blockID flow.Identifier, height uint64) error {
+			return nil
+		},
+		TransactionsFunc: func(height uint64, transactions []*flow.TransactionBody) error {
+			return nil
+		},
+	}
+
+	feeder := &mocks.Feeder{
+		UpdateFunc: func() (*ledger.TrieUpdate, error) {
+			return testUpdate, nil
+		},
+	}
+
+	once := &sync.Once{}
+
+	forest := &mocks.Forest{
+		SaveFunc: func(tree *trie.MTrie, paths []ledger.Path, parent flow.StateCommitment) {},
+		HasFunc: func(commit flow.StateCommitment) bool {
+			return true
+		},
+		TreeFunc: func(commit flow.StateCommitment) (*trie.MTrie, bool) {
+			return testTrie, true
+		},
+		PathsFunc: func(commit flow.StateCommitment) ([]ledger.Path, bool) {
+			return testPaths, true
+		},
+		ParentFunc: func(commit flow.StateCommitment) (flow.StateCommitment, bool) {
+			return testLastCommit, true
+		},
+		ResetFunc: func(finalized flow.StateCommitment) {},
+		SizeFunc: func() uint {
+			return 42
+		},
+	}
+
+	registers := map[ledger.Path]*ledger.Payload{
+		testPath1: testPayload1,
+		testPath2: testPayload2,
+		testPath3: testPayload3,
+		testPath4: testPayload4,
+		testPath5: testPayload5,
+		testPath6: testPayload6,
+	}
```

Comment:
For me, the baseline FSM should not have any collected registers in it. That's state that is added after initialization, and only for one specific state.